### PR TITLE
Script to unbind and bind the Network virtualized device

### PR DESCRIPTION
**Pull request by hathyaga**
This script will unbind and bind the driver of the
Network virtualized device based on the count given by
the user

Signed-off-by: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by hathyaga** [comment]
Job log
[job.log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1819157/job.log.txt)

File: io/net/Network_virtualization_driver_bind.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel("%s package is need to test" % pkg)
+        interfaces = netifaces.interfaces()
+        self.interface = self.params.get('interface')
+        if self.interface not in interfaces:
+            self.cancel("%s interface is not available" % self.interface)
+        self.device = process.system_output("ls -l /sys/class/net/ | \
+                                             grep %s | cut -d '/' -f \
+                                             5" % self.interface,
+                                            shell=True).strip()
```
[the remark this answers is not in the document]
Comment (hathyaga) [reply io/net/Network_virtualization_driver_bind.py]: Here is where I am finding the device ID of the Networl virtualized interface.

[the remark this answers is not in the document]
Comment (narasimhan-v) [reply io/net/Network_virtualization_driver_bind.py]: @hathyaga if you can give me examples, I can point you to an API in avocado which does that.

**Comment by hathyaga** [comment]
Why not have this as a separate test to be included under tests for Network virtualized device? 
I agree it can be merged under io/driver/driver_bind_unbind test But I need this categorically put under Network virtualized device tests.

**Comment by narasimhan-v** [comment]
What is the justification ?
This test only differs in how we find the interface and device ID. But no different in the actual test that driver bind unbind test does.

**Comment by hathyaga** [comment]
That in itself is the justification. That this test is performed for an non pci device and hence requesting this to be pushed as new test.

File: io/net/Network_virtualization_driver_bind.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hathyaga) [reply io/net/Network_virtualization_driver_bind.py]: (1) root @ CI-s822l1-lp3: /root
# ls -l /sys/class/net/
total 0
lrwxrwxrwx. 1 root root 0 Mar 14 20:08 eth0 -> ../../devices/vio/30000003/net/eth0
lrwxrwxrwx. 1 root root 0 Mar 14 16:24 lo -> ../../devices/virtual/net/lo
lrwxrwxrwx. 1 root root 0 Mar 14 16:24 net0 -> ../../devices/vio/30000002/net/net0

Where eth0 is the Network virtualized device

**Comment by narasimhan-v** [comment]
There is an agreement that this is a non-pci device. And that is not under discussion here.

This is a driver based test, and should be treated like one, IMHO. If there is a possibility to treat it in a generic way, it should be done so.

File: io/net/Network_virtualization_driver_bind.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        self.count = int(self.params.get('count', default="1"))
+
+    def test(self):
+        '''
+        Performs driver unbind and bind for the Network virtualized device
+        '''
+        path = "/sys/devices/vio/%s/driver" % self.device
```
Comment (narasimhan-v) [inline io/net/Network_virtualization_driver_bind.py]: shouldn't it be `/sys/bus/vio/drivers/%s` ?

Comment (hathyaga) [reply io/net/Network_virtualization_driver_bind.py]: No. The correct path is as seen below which is what is used in the script as well.

ltc-zz1b-lp11:~ # ll /sys/class/net/ | grep -i eth7
lrwxrwxrwx 1 root root 0 Mar 20 02:18 eth7 -> ../../devices/vio/30000003/net/eth7
ltc-zz1b-lp11:~ # cd /sys/devices/vio/30000003
ltc-zz1b-lp11:/sys/devices/vio/30000003 # ls
devspec  driver  failover  modalias  name  net  of_node  power  subsystem  uevent
ltc-zz1b-lp11:/sys/devices/vio/30000003 # cd driver
ltc-zz1b-lp11:/sys/devices/vio/30000003/driver # ls
30000003  bind  module  uevent  unbind

**Comment by hathyaga** [comment]
Agreed that there is a generic test for this test but there can be multiple implementations of the existing test and this is one example for that. 
I bring in the case of this being specific to non-pci devices and hence will want to push this test to virt-net

**Comment by hathyaga** [comment]
New PR: https://github.com/avocado-framework-tests/avocado-misc-tests/pull/828

**Pull request closed without merging**